checkstyle.xml

```diff
@@ -2,47 +2,219 @@
+		<module name="WhitespaceAfter">
+			<property name="tokens" value="COMMA,SEMI,TYPECAST,LITERAL_ELSE,LITERAL_DO"/>
+		</module>
+		
+		<!-- Ensure that there are whitespace around the given tokens -->
+		<module name="WhitespaceAround">
+			<property name="allowEmptyConstructors" value="true"/>
+			<property name="allowEmptyMethods" value="true"/>
+			<property name="allowEmptyTypes" value="true"/>
+			<property name="allowEmptyLoops" value="true"/>
+			<property name="allowEmptyLambdas" value="true"/>
+			<property name="allowEmptyCatches" value="true"/>
+			<property name="tokens" value="ASSIGN, BAND, BAND_ASSIGN, BOR, BOR_ASSIGN, BSR, BSR_ASSIGN, BXOR,
+				BXOR_ASSIGN, COLON, DIV, DIV_ASSIGN, EQUAL, GE, GT, LAMBDA, LAND, LCURLY, LE, LITERAL_CATCH, LITERAL_DO,
+				LITERAL_ELSE, LITERAL_FINALLY, LITERAL_RETURN, LITERAL_TRY,
+				LOR, LT, MINUS, MINUS_ASSIGN, MOD, MOD_ASSIGN, NOT_EQUAL, PLUS, PLUS_ASSIGN, QUESTION,
+				RCURLY, SL, SLIST, SL_ASSIGN, SR, SR_ASSIGN, STAR, STAR_ASSIGN, TYPE_EXTENSION_AND"/>
+		</module>
+		
+		<!-- Ensure that line-wrapped operators appear after the newline -->
+		<module name="OperatorWrap">
+			<property name="tokens" value="QUESTION, COLON, EQUAL, NOT_EQUAL, DIV, PLUS, MINUS, STAR, MOD, SR, BSR, GE,
+				GT, SL, LE, LT, BXOR, BOR, LOR, BAND, LAND, LITERAL_INSTANCEOF, TYPE_EXTENSION_AND"/>
+		</module>
+		
+		<!-- Ensure that overloaded methods are grouped together (have no other method in between) -->
+		<module name="OverloadMethodsDeclarationOrder"/>
+		
+		<!-- Ensure package name format. Default format: "^[a-z]+(\.[a-zA-Z_][a-zA-Z0-9_]*)*$" -->
+		<module name="PackageName"/>
+		
+		<!-- Ensure method and catch parameter name format. Default format: "^[a-z][a-zA-Z0-9]*$" -->
+		<module name="ParameterName"/>
+		
+		<!-- Disallow whitespaces after '(' and before ')' -->
+		<module name="ParenPad"/>
+		
+		<!-- Disallow redundant (double or default-available class) imports -->
+		<module name="RedundantImport"/>
+		
+		<!-- Require "this" for non-static field access -->
+		<module name="RequireThis"/>
+		
+		<!-- Ensure usage of "}" for the given tokens -->
+		<module name="RightCurly">
+			<property name="tokens" value="LITERAL_TRY, LITERAL_CATCH, LITERAL_FINALLY, LITERAL_IF, LITERAL_ELSE"/>
+		</module>
+		
+		<!-- Ensure that when lines are wrapped at a dot, this dot is on the new line. -->
+		<module name="SeparatorWrap">
+			<property name="option" value="nl"/>
+			<property name="tokens" value="DOT"/>
+		</module>
+		
+		<!-- Ensure that when lines are wrapped at a comma, this comma is at the end of the line. -->
+		<module name="SeparatorWrap">
+			<property name="option" value="eol"/>
+			<property name="tokens" value="COMMA"/>
+		</module>
+		
+		<!-- Ensure non-static, non-final field name format. Default format: "^[a-z][a-zA-Z0-9]*$" -->
+		<module name="StaticVariableName"/>
+		
+		<!-- Disallow the "==" operator for string literals -->
+		<module name="StringLiteralEquality"/>
```

Comment:
There may be a few places where this is a problem, because I am validly using == in place of .equals, so I'm curious to see the results of this. Certainly though, in 90% of the cases, this will be an error in the code, so perhaps we just do //CHECKSTYLE:OFF on those blocks.

Reply:
This should only fire when a string literal (and not a variable) is checked using "==". It is still possible to perform `someVar == someOtherVar`, but not `someVar == "str"` (which would always be false anyways).

Reply:
@LadyCailin This rule currently has one violation in `BukkitBlockEvents:447`, for code : `item.getType().getName() == "AIR"`, which is luckily only used as a condition to throw a CRE in a very specific situation at the moment, but it definately is a mistake which could lead to a failure/bug.